server/svix-server/src/v1/endpoints/message.rs

```diff
@@ -154,12 +158,35 @@ async fn list_messages(
 async fn create_message(
     Extension(ref db): Extension<DatabaseConnection>,
     Extension(queue_tx): Extension<TaskQueueProducer>,
+    Extension(redis_cache): Extension<Option<RedisCache>>,
     ValidatedJson(data): ValidatedJson<MessageIn>,
     AuthenticatedApplication { permissions, app }: AuthenticatedApplication,
 ) -> Result<(StatusCode, Json<MessageOut>)> {
     let txn = db.begin().await?;
     let db = &txn;
 
+    let create_message_app: CreateMessageApp = if let Some(redis_cache) = redis_cache {
+        let cache_key = AppEndpointKey::new(app.org_id.clone(), app.id.clone());
+        // Redis cache errors are logged but ignored because you can always fallback to postgres
+        if let Some(cma) = redis_cache.get(&cache_key).await.unwrap_or_else(|e| {
+            tracing::error!("Redis cache error on fetch: {}", e);
+            None
+        }) {
+            cma
+        } else {
+            // If the [`CreateMessageApp`] isn't in the Redis cache, fetch it from the PostgreSQL
+            // database and insert that into the cache
+            let cma = CreateMessageApp::fetch(db, app.clone()).await?;
+            if let Err(e) = redis_cache.set(&cache_key, &cma, Some(30)).await {
+                tracing::error!("Redis cache error on set: {}", e);
+            }
+            cma
+        }
+    } else {
+        // If Redis is disabled, there will be no RedisCache so fallback to postgres
+        CreateMessageApp::fetch(db, app.clone()).await?
+    };
```

Comment:
This looks like it should just be a function that does all of this, and is shared between the worker and the message.

I realize that in `create_message` we already have the `app` fetched, so we can just pass an optional `app` parameter to not fetch the app if it exists, but other than that, they should both be identical.